src/main/java/org/medicmobile/webapp/mobile/MedicAndroidJavascript.java

```diff
@@ -230,6 +246,58 @@ public void sms_send(String id, String destination, String content) throws Excep
 		}
 	}
 
+	@org.xwalk.core.JavascriptInterface
+	@android.webkit.JavascriptInterface
+	public String getDeviceInfo() {
+		try {
+			if (activityManager == null) {
+				return jsonError("ActivityManager not set.  Cannot retrieve RAM info.");
+			}
+
+			String androidVersion = Build.VERSION.RELEASE;
+			int osApiLevel = Build.VERSION.SDK_INT;
+			String osVersion = System.getProperty("os.version") + "(" + android.os.Build.VERSION.INCREMENTAL + ")";
+			String device = Build.DEVICE;
+			String model = Build.MODEL;
+			String manufacturer = Build.BRAND;
+			String harware = Build.HARDWARE;
+
+			File dataDirectory = Environment.getDataDirectory();
+			StatFs dataDirectoryStat = new StatFs(dataDirectory.getPath());
+			long dataDirectoryBlockSize = dataDirectoryStat.getBlockSizeLong();
+			long dataDirectoryAvailableBlocks = dataDirectoryStat.getAvailableBlocksLong();
+			long dataDirectoryTotalBlocks = dataDirectoryStat.getBlockCountLong();
+			String freeMemorySize = formatSize(dataDirectoryAvailableBlocks * dataDirectoryBlockSize);
+			String totalMemorySize = formatSize(dataDirectoryTotalBlocks * dataDirectoryBlockSize);
```

Comment:
Can we also add Runtime.maxMemory? This returns the maximum amount of memory that the Java virtual machine will attempt to use.